### DISCUSSION TITLE
dropDown is not able to find select menu by label

### DISCUSSION THREAD

**Discussion by leandrosansilva** [getgauge/taiko#1729] [General]
**Describe the bug**

**To Reproduce**
Steps (or script) to reproduce the behavior:

This is the static HTML for the page in the link used below (https://leandrosansilva.gitlab.io/taiko-page-test/):

```html
<html>
  <body>
    <select>
      <option value="" selected="selected" disabled="disabled">Most of my mail is…</option>
      <option value="direct">Direct (personal, office, one-to-one)</option>
      <option value="transactional">Transactional (notifications, apps)</option>
      <option value="marketing">Marketing (newsletters, adverts)</option>
    </select>
  </body>
</html>
```

For the context, in the real application, the tags attributed (selected, disabled) are generated by Vue.js, and the issue might lie there.

This is the taiko code:

```js
const { openBrowser, write, closeBrowser, goto, dropDown } = require('taiko');
openBrowser();
goto('https://leandrosansilva.gitlab.io/taiko-page-test/');
dropDown("Most of my mail is…").select("marketing"); // this line fails as the drop down menu is not found
```

But it kind of works if I omit the label (I assume taiko uses the first found element):

```js
dropDown().select("marketing");
```

**Expected behavior**
I expect taiko to find the menu and select the option "marketing".

**Screenshot**
![image](https://user-images.githubusercontent.com/1503334/101884603-434bac80-3b99-11eb-9181-79e035b3271f.png)

**Versions:**
 - Taiko: Version: 1.1.0 (Chromium: 88.0.4314.0) RELEASE
 - OS Manjaro Linux (latest as on Dec 2020) amd64 (On KDE)
 - Node.js v15.3.0

Gauge version: 1.1.6
Commit Hash: 3af456f

Plugins
-------
html-report (4.0.12)
js (2.3.12)
screenshot (0.0.1)
xml-report (0.2.3)

**Comment by NivedhaSenthil** [comment]
`dropDown` api expects a label to identify the select tag element..in the example you have mentioned it is an option which you were trying to use

**Comment by leandrosansilva** [comment]
> `dropDown` api expects a label to identify the select tag element..in the example you have mentioned it is an option which you were trying to use

Hi @NivedhaSenthil thank you for the quick reply. I've realized I ruched and misnamed the issue and just fixed its title.

Now the title describes the issue properly.

I'd expect `dropDown` to find my menu by how it looks visually. In case my assumption is invalid, what would you recommend me to use instead?

**Comment by zabil** [comment]
Since there is only one dropdown on your page the following should work

```
dropDown().select("marketing"); 
```

As mentioned earlier "Most of my mail is…" is a dropdown value and not a label.

**Comment by zabil** [comment]
Also to match part text you should ideally be using regex support

```
dropDown().select(/Marketing/); 
```

**Comment by leandrosansilva** [comment]
> Since there is only one dropdown on your page the following should work
> 
> ```
> dropDown().select("marketing"); 
> ```
> 
> As mentioned earlier "Most of my mail is…" is a dropdown value and not a label.

Yes, I mentioned the empty `label` as it obviously works in my example, but won't in case I have multiple dropdowns in a page, where probably the first one will always be chosen.

@zabil would you recommend another way of finding the select by the visually displayed selected option?

**Comment by zabil** [comment]
The only other way is to refer the dropdown by index https://docs.taiko.dev/working_with_element_lists/#using-indexes-and-loops

Or use a proximity selector to locate the dropdown near another element

```
await dropDown(below('text')).exists()
```

**Reply by leandrosansilva** [reply]
Thx, I'll try it.
